File: fix_ts.js
const fs = require('fs');
const path = './src/modules/orders/orders.service.ts';
let code = fs.readFileSync(path, 'utf8');

// Fix 1: sendPushNotification to sendNotification
code = code.replace(/this\.notificationsService\.sendPushNotification\([^]+?\}\s*\)\.catch\(\(\) => \{\}\);/g, (match) => {
  return match.replace(/sendPushNotification\(\s*([^,]+),\s*([^,]+),\s*([^,]+),\s*([^)]+)\s*\)/, 'sendNotification($1, { title: $2, body: $3, payload: $4 })');
});

// Fix 2: sendNotificationToUser to sendToUser
code = code.replace(/this\.notificationsGateway\.sendNotificationToUser/g, 'this.notificationsGateway.sendToUser');

// Fix 3: any cast for vendor and price
code = code.replace(/if \(substituteObj\.vendor\?/g, 'if ((substituteObj as any).vendor?');
code = code.replace(/substituteObj\.price/g, '(substituteObj as any).price');

// Also fix the payload of sendToUser which expected { type, title, message, payload } but sendToUser expects { title, body, type, data }
code = code.replace(/sendToUser\([^]+?\}\);/g, (match) => {
  let newMatch = match.replace('message:', 'body:').replace('payload:', 'data:');
  return newMatch;
});

fs.writeFileSync(path, code);

File: fix_ts.py
import re

with open("src/modules/orders/orders.service.ts", "r") as f:
    content = f.read()

# Add HttpException import if missing
if 'HttpException' not in content:
    content = content.replace("NotFoundException,", "NotFoundException, HttpException,")
    # or just find "import { " and add it if we can't find exact match, but let's just do:
    content = re.sub(r'import \{ (.*?) \} from \'@nestjs/common\';', r'import { \1, HttpException } from \'@nestjs/common\';', content, count=1)

# Fix substituteOptions type
content = content.replace('const substituteOptions = [];', 'const substituteOptions: any[] = [];')

# Fix substituteObj.price casting
content = content.replace('price: substituteObj.price', 'price: (substituteObj as any).price')
content = content.replace('docItem.price = substituteObj.price', 'docItem.price = (substituteObj as any).price')
content = content.replace('docItem.subtotal = substituteObj.price * quantity', 'docItem.subtotal = (substituteObj as any).price * quantity')
content = content.replace('const newSubtotal = substituteObj.price * quantity', 'const newSubtotal = (substituteObj as any).price * quantity')

with open("src/modules/orders/orders.service.ts", "w") as f:
    f.write(content)

